### PR TITLE
fix: resolve the accessibility issues pointed out for header component

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#154 

#### What's this PR do?
- Fixes accessibility issues with header
- Fixes dropdown button unique id issue highlighted by LightHouse
- Fixes area-label texts
- Fixes `alt` for MIT logo in the header

#### How should this be manually tested?
- Check the accessibility on any page and there should be no complaints regarding the header part.

#### Screenshots (if appropriate)

**After**
Screenshot 2021-09-09 at 6 06 54 PM
Screenshot 2021-09-09 at 6 08 15 PM

**Before**
Screenshot 2021-09-09 at 6 11 36 PM

### CODE REVIEW

File: static/js/components/AnonymousMenu.js [hunk 1]
```diff
@@ -22,15 +22,17 @@ const AnonymousMenu = ({ useScreenOverlay }: Props) => {
...
           className="simple"
-          aria-label="Login"
+          aria-label="Sign In"
         >
           Sign In
         </MixedLink>
       </li>
       <li {...listItemProps || {}}>
         <MixedLink
+          id="create_account"
```
Comment (gsidebo) [inline static/js/components/AnonymousMenu.js:35]: Nit: I think we want to stick with camel case for id/name attributes

```suggestion
          id="createAccount"
```

**Review by gsidebo — APPROVED** [review]
Looks great! 👍  You can take or leave my suggestion. Feel free to merge

**Comment by gsidebo** [comment]
@arslanashraf7 Before you merge, can you change the "What are the relevant tickets?" section from `#154` to `Closes #154`? Please do that for all future PRs too

**Comment by arslanashraf7** [comment]
> @arslanashraf7 Before you merge, can you change the "What are the relevant tickets?" section from `#154` to `Closes #154`? Please do that for all future PRs too

@gsidebo I used to do this in the past and I know it will close the associated ticket and move it to `Done` once the PR is merged. I don't remember the place exactly but I believe there has been some conversation about when we consider a ticket to be Done which is once a ticket has made it to the end-user(production). 

However, I can start following that again if that's the standard practice we're still following.

@pdpinch what's your opinion on the above?

**Comment by asadiqbal08** [comment]
I have the same question as @arslanashraf7 raised, Thanks for asking.

File: static/js/components/AnonymousMenu.js (hunk 1 above)
Comment (arslanashraf7) [reply static/js/components/AnonymousMenu.js]: @gsidebo, Thanks for the suggestion, I've converted it to the camel case and in the process, I did a quick refactoring in the AnonymousMenu as well, similar to what I did in UserMenu in this PR. 

Here is the commit https://github.com/mitodl/mitxonline/pull/168/commits/22dc55ab142fd8bcd13603238b3793e6e1dc3de5, please let me know if you'd like any changes or this looks good as well.

**Review by gsidebo — APPROVED** [review]
Looks great! 👍

**Pull request merged**